### PR TITLE
Fix require edge cases

### PR DESCRIPTION
**Pull request by platypii**
Very small PR that fixes two issues:

1. Fix `clearCache` function which was doing nothing, due to misuse of `Cache.keys()` function.

2. Fix regression on relative imports.  For example, import `./brother.js` from `src/sister.js` was missing `src` from base path, and so it would look in the top level directory instead of the `src` folder. This worked previously but was regressed by #51.

`index.js`:
```js
const bro = require('./src/brother.js')
```

`src/brother.js`:
```js
const sis = require('./sister.js')
```

`src/sister.js`:
```js
module.exports = { main: () => [] }
```

### CODE REVIEW

File: packages/require/src/require.js [hunk 1]
```diff
@@ -59,8 +59,8 @@ export function require(urlOrSource, transform, _readFile=readFileWeb, _base, ro
 
     if (isModule) {
       readFile = readModule
-      base = root = url + '/'
     }
+    base = url
```
Comment (hrgdavor) [inline packages/require/src/require.js:63]: not sure if this will break something else in that long line of imports of jscad/io. I will check it out again with this line included.

Comment (hrgdavor) [reply packages/require/src/require.js:63]: looks it works after disabling jscadio bundle, al hose funky imports are ok.

**Review by hrgdavor — APPROVED** [review]
looks ok. thnx-

**Comment by platypii** [comment]
What started as a 2-line PR turned into something bigger. But I **THINK** this fixes all the the weird edge cases in terms of package management. Hopefully :crossed_fingers: 

Changes since the original PR:

 - Add comments and jsdoc
 - fs-provider: split out `getWorkspaceAliases` function. I also slightly refactored it to return a list of `{ name, path }` objects instead of array of arrays to make the code clearer and more robust.
 - `_readFile` can't have a default value since its in the middle of the args.
 - remove `readModule` parameter since its always equal to `readFile`.
 - use `resolvedUrl` instead of overwriting `url` variable. easier to understand what's happening.
 - use `resolvedUrl` for `cacheUrl`. this simplifies the logic, and ensures that we always cache on the most-resolved, canonical url.
 - remove `base` from `readFile` calls, since `resolveUrl` already did the work of constructing the URL.
 - fix require package tests
    
This fixes edge cases including:
 - relative imports from `"main": "src/index.js"`
 - workspace imports
 - better error messages on failed imports

I tested on a TON of different project structures, although I'm sure there's things I missed. Let me know what you think @hrgdavor

**Review by hrgdavor — APPROVED** [review]
I confirm that it works with my 3d print mega project with workspaces. `readModule ` was a bit of overthinking, it might be needed in the future, or not(I had some vague idea I will need it as separate option).  Thnx for the refactor and improvements. This code for runing js projects will go beyond just jscad for me.
